postgres-types/src/lib.rs

```diff
@@ -612,14 +613,26 @@ impl<'a> FromSql<'a> for Box<str> {
 }
 
 impl<'a> FromSql<'a> for &'a str {
-    fn from_sql(_: &Type, raw: &'a [u8]) -> Result<&'a str, Box<dyn Error + Sync + Send>> {
-        types::text_from_sql(raw)
+    fn from_sql(ty: &Type, raw: &'a [u8]) -> Result<&'a str, Box<dyn Error + Sync + Send>> {
+        match *ty {
+            ref ty if (
+                ty.name() == "ltree"    ||
+                ty.name() == "lquery"   ||
+                ty.name() == "ltxtquery"
+            ) => types::ltree_from_sql(raw),
```

Comment:
Are these types guaranteed to use the same encoding? It may be better to expose separate to_sql and from_sql methods in postgres_protocol for each even if they currently do the same thing.

Reply:
Split out to/from for each type